giraffe/src/utils/getMargins.ts

```diff
@@ -13,16 +13,33 @@ export const getMargins = (
   yAxisLabel: string,
   yTicks: number[],
   yTickFormatter: (tick: number) => string,
-  tickFont: string
+  tickFont: string,
+  yColumnLabels: string[]
 ) => {
   if (!showAxes) {
     return {top: 1, right: 1, bottom: 1, left: 1}
   }
 
-  const longestYTick = maxBy(
-    d => d.length,
-    yTicks.map(t => yTickFormatter(t))
-  )
+  let longestYTick
+  if (yColumnLabels) {
+    // for cases where they y-axis has strings as values
+    // find the longest label in the list
+    let maxVal = yColumnLabels[0]
+    let maxLength = maxVal.length
+    let i
+    for (i = 1; i < yColumnLabels.length; i++) {
```

Comment:
it looks like we want to use the function [maxBy](https://github.com/influxdata/giraffe/blob/master/giraffe/src/utils/extrema.ts#L21) similar to below. It seems like it's made for this. Of course it's hard to tell because the function is dense and looks like mathematical notation. After staring at it for a good 10 minutes, I *think* this is what it do:

The first argument is a function that gets run on every item of the array. That function should produce the value you're comparing. In this case, length of the label. The second argument is the array to get the max value for.

I **think** (haven't tested it) you want something like this:

```ts
  let longestYTick
  if (yColumnLabels) {
    longestYTick = maxBy(
      d => d.length,
      yColumnLabels
    )
  }
```